src/components/DappBrowser/Homepage.tsx

```diff
@@ -186,6 +190,22 @@ const Card = React.memo(function Card({
     }
   }, [addFavorite, isFavorite, removeFavorite, site]);
 
+  const hasVisited = useBrowserHistoryStore(state => state.hasVisited);
+  const dappClickedBefore = useMemo(() => hasVisited(site.url), [hasVisited, site.url]);
```

Comment:
instead of creating this, you can simply call `hasVisited(site.url)` within `handlePress`

Reply:
cool. didn't know if there would be any issues with hasVisited having the correct state data that way.

Reply:
hmm. i am not actually sure if this works. When it's memorized, the `hasClickedBefore` updates as expected (when a trending dapp is visited for the second time), but when i pass `hasVisited(site.url)` directly in it only flips `hasClickedBefore` to `true` after the third visit.